glean-core/ios/GleanTests/HttpPingUploaderTests.swift

```diff
@@ -0,0 +1,167 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import XCTest
+import OHHTTPStubs
+@testable import Glean
+
+class HttpPingUploaderTests: XCTestCase {
+    var expectation: XCTestExpectation?
+    private let testPath = "/some/random/path/not/important"
+    private let testPing = "{ \"ping\": \"test\" }"
+    private let testConfig = Configuration(
+        userAgent: "Glean/Test 25.0.2",
+        connectionTimeout: 3050,
+        logPings: true,
+        pingTag: "Tag"
+    )
+
+    private func setupHttpResponseStub(statusCode: Int32 = 200) {
+        stub(condition: isHost("incoming.telemetry.mozilla.org")) { data in
+            let body = (data as NSURLRequest).ohhttpStubs_HTTPBody()
+            let str = String(data: body!, encoding: .utf8) ?? ""
+            print("STUB RECEIVED: \(str) \n")
+            // swiftlint:disable force_try
+            let json = try! JSONSerialization.jsonObject(with: body!, options: []) as? [String: Any]
+            XCTAssert(json != nil)
+            XCTAssertEqual(json?["ping"] as? String, "test")
+
+            return OHHTTPStubsResponse(
+                jsonObject: ["foo": "bar"],
```

Comment:
Why do we return a `foo: bar` here?

Reply:
Because it looks like the only functions that allow for this require either a file or data or a JSON object to be returned.  This is something I borrowed from the [telemetry-ios](https://github.com/mozilla-mobile/telemetry-ios/blob/d092de49d715d0bd1c6fad4bd89c85e51172d444/TelemetryTests/TelemetryTests.swift#L71) library.

Reply:
Can we just return an empty JSON object, i.e. `{}`?

Reply:
@mdboom yes, that's probably the easiest solution since we aren't using the response body for anything.